### ISSUE TITLE
Autosave not restoring windows

### ISSUE THREAD

**Issue by dkyy** [KrabCode/LazyGui#305]
Hello. I just updated to 1.8.3 from something like 1.3.? and noticed that when I launch my project the autosave is correctly restoring the gui's values - but none of the gui windows I had opened are restored (ie. they remain closed/hidden). 
I've tried removing the auto.json file - but that didn't fix the problem.
Does anyone have a suggestion for what I might do to correct this problem?
Thanks,

david,

**Comment by dkyy** [comment]
I just noticed that none of the json save files have any "posX" or "posY" values stored.

**Comment by KrabCode** [comment]
Yeah - remembering and restoring window state is no longer a thing,
read more here: https://github.com/KrabCode/LazyGui/issues/273

did you have some specific use case in mind? do you want this functionality restored?

**Comment by dkyy** [comment]
Yes - remembering and restoring window state is incredibly useful for me.

In my use case I often quit my Processing application, make some adjustments to my code, and restart. And having my GUI windows restored allows me to get back to my previous state incredibly quickly and easily. Otherwise I need re-open multiple gui windows, rearrange them, etc, before I can get back to work. That process slows my workflow and is distracting - I can forget the details of what I was doing.

Perhaps you could add an option to allow remembering and restoring?

Thanks so much,

david,

p.s. As a side note I switched to LazyGui from Interfascia about a year or two ago, and it's been a revelation. I'm able to do so much more, and so much more quickly - I'm a huge fan of this project!

**Comment by dkyy** [comment]
I'm not sure I understand the issue of windows rearranging -- I never had that problem.
But if you can get something working I'd be super grateful.
Thanks!

**Comment by KrabCode** [comment]
The windows rearranging were a problem because for example:

you know how there's a "saves" folder in the gui at runtime and that "saves" window also used to remember its position per save

imagine you clicked it in order to load a save and the whole window you were just using moved elsewhere - super jarring

**Comment by KrabCode** [comment]
I finally got around to coding something for this - I seem to have restored this functionality pretty easily, but now I need to take a while to decide on a good list of options and a sensible default value. 

I'm thinking there should be three options in the constructor settings:
  A) only the first auto-load will restore windows
  B) all loading will restore windows
  C) no restoring of windows at all

I'd be most comfortable with A as the default honestly.

Side note for the "saves" folder: I added an extra rule for this particular built-in folder; its window will always be completely unaffected by loading saves so that it's not jarring to see it disappear from under you and it's never really an essential part of controlling a sketch, so I don't think anyone will mind that it won't be restored on startup.
